### ISSUE TITLE
Builder: Restructure initConfig function

### ISSUE THREAD

**Issue by Chinwendu20** [open-telemetry/opentelemetry-collector#6955]
Move this [line](https://github.com/open-telemetry/opentelemetry-collector/blob/f2a0133bf677b9fe2e18b7c02eb1b6bf5e849999/cmd/builder/internal/command.go#L158) into this [conditional block](https://github.com/open-telemetry/opentelemetry-collector/blob/f2a0133bf677b9fe2e18b7c02eb1b6bf5e849999/cmd/builder/internal/command.go#L130)

**Comment by gbbr** [comment]
The problem with this change is that if you were to do it, you would see the log line even if the config fails to load. Whereas currently you only see it after the config has successfully loaded. I think that was the initial intention to begin with. The `initConfig` function has [multiple](https://github.com/open-telemetry/opentelemetry-collector/blob/f2a0133bf677b9fe2e18b7c02eb1b6bf5e849999/cmd/builder/internal/command.go#L140) [return](https://github.com/open-telemetry/opentelemetry-collector/blob/f2a0133bf677b9fe2e18b7c02eb1b6bf5e849999/cmd/builder/internal/command.go#L147) [points](https://github.com/open-telemetry/opentelemetry-collector/blob/f2a0133bf677b9fe2e18b7c02eb1b6bf5e849999/cmd/builder/internal/command.go#L152) before that line.

**Comment by Chinwendu20** [comment]
Thanks @gbbr . My thought process was that there is logger info when the default config is used here:
https://github.com/open-telemetry/opentelemetry-collector/blob/f2a0133bf677b9fe2e18b7c02eb1b6bf5e849999/cmd/builder/internal/command.go#L136
In which case there could be a possibility of a failure to load the default config file. Then the other line for custom config could be moved upwards as well.

Or for the sake of uniformity we could move both down after all the possible errors, if that makes sense

**Comment by gbbr** [comment]
You're right! I did not notice this detail. I guess in that case we should make it consistent and have them both in the same place.

**Comment by jpkrohling** [comment]
> Or for the sake of uniformity, we could move both down after all the possible errors if that makes sense

I'm happier with this proposal.
